lib/remote/apilistener-filesync.cpp

```diff
@@ -330,7 +330,7 @@ void ApiListener::HandleConfigUpdate(const MessageOrigin::Ptr& origin, const Dic
 	/* Only one transaction is allowed, concurrent message handlers need to wait.
 	 * This affects two parent endpoints sending the config in the same moment.
 	 */
-	auto lock (Shared<std::unique_lock<SpinLock>>::Make(m_ConfigSyncStageLock));
+	std::lock_guard<std::mutex> lock(m_ConfigSyncStageLock);
```

Comment:
Why not `std::unique_lock<>`?

Reply:
Because `lock_guard` suffices, but `unique_lock` should also be totally fine, it just provides more than what is needed here. Do you want me to change it?

Reply:
More than what is needed here? What exactly?

Reply:
All the additional functionality that `unique_lock` provides but is not needed: deferred locking, try lock, early unlocking, etc.